### PR TITLE
sagemath: update to 10.2, flintlib: update to 3.0.1

### PR DESCRIPTION
**Pull request by tornaria**
#### Testing the changes
- I tested the changes in this PR: **YES**

This is a major update for both sagemath and flintlib.

Sagemath moves to cython 3.

A couple of packages need rebuild or update:

 - eclib: rbuild for flintlib-3.0.1
 - singular: update to 4.3.2p10
 - python3-pplpy: rebuild using cython 3
 - python3-networkx: update to 3.2.1.

### CODE REVIEW

**Comment by dkwo** [comment]
Excellent. This builds and checks fine on `aarch64`, after rebasing.

**Comment by tornaria** [comment]
@dkwo sagemath is on its second RC so it will be released soon. This PR will be ready to merge once the release happens. It should ideally be tested together with #46785 and #46787.

**Comment by dkwo** [comment]
possibly eclib as well?

**Comment by tornaria** [comment]
> Can flint (and possibly sage) be built without ntl after this update? i read somewhere that ntl might not be necessary anymore, and this would ease the burden of cross compiling.

I think flintlib is already cross-compiled.

As for eclib, it lists `ntl-devel` in makedepends. Have you tried building it cross without `ntl-devel`? If it works, we can just add a build option ntl that is disabled by default in cross builds. However, I think that `eclib` actually uses `ntl` for some things.

As for sagemath, I think it has a few more nocross makedepends. The comment in the template list some; including flintlib, but I'm not sure this is a mistake. It's possible that sagemath needs a flintlib that was built with ntl.

File: srcpkgs/sagemath/template [hunk 1]
```diff
@@ -33,7 +33,7 @@ license="GPL-2.0-or-later"
 homepage="https://www.sagemath.org/"
 changelog="https://github.com/sagemath/sage/releases"
 distfiles="https://github.com/sagemath/sage/archive/refs/tags/$version.tar.gz"
-checksum=a658612b1b2376ddaf207cc8ed0ef458d4c2880c16e19139bedbe8baa42ad62f
+checksum=d13f4bdc5db4f8602be0453c51765a563f6ff5c4cf2d80c696fbb2194d0932ce
 nocross="due to ntl (flintlib), fflas-ffpack, givaro, linbox, cysignals, sympow, maxima"
```
Comment (dkwo) [inline srcpkgs/sagemath/template:37]: could you remove cysignals and add eclib (again due to ntl)?

**Comment by dkwo** [comment]
The current iteration (rc4) has a couple of test failures on aarch64:
```
sage/libs/gap/element.pyx (killed due to segfault)
sage/matrix/matrix_integer_dense.pyx (killed due to segfault)
sage/graphs/generic_graph.py (timed out)
```
With regard to the crossbuild, I will investigate eclib and flint (in the past, our crossbuilt flintlib was giving a doctest error in sage).

**Comment by dkwo** [comment]
(previously, I was testing with `XBPS_CHECK_PKGS=yes`, yesterday I switched to `full`.)

File: srcpkgs/sagemath/template (hunk 1 above)
Comment (dkwo) [reply srcpkgs/sagemath/template:37]: as well as remove flintlib, per discussion in the sage thread?

**Comment by tornaria** [comment]
> The current iteration (rc4) has a couple of test failures on aarch64:
> 
> ```
> sage/libs/gap/element.pyx (killed due to segfault)
> sage/matrix/matrix_integer_dense.pyx (killed due to segfault)
> sage/graphs/generic_graph.py (timed out)
> ```
> 
> With regard to the crossbuild, I will investigate eclib and flint (in the past, our crossbuilt flintlib was giving a doctest error in sage).

Can you try running those tests again? At least the `matrix_integer_dense.pyx` I think I've seen from time to time as a random segfault. The others I don't know, if you can reproduce consistently, then try to see what triggers it, get a minimal example that reproduces the issue, and submit a bug report upstream.

**Comment by tornaria** [comment]
@dkwo this is ready to merge, any comments?

**Pull request merged**